src/snowflake/snowpark/_internal/packaging_utils.py

```diff
@@ -28,6 +30,90 @@
 }
 
 
+def parse_requirements_text_file(file_path: str) -> Tuple[List[str], List[str]]:
+    """
+    Parses a requirements.txt file to obtain a list of packages and file/folder imports. Returns a tuple of packages
+    and imports.
+    Args:
+        file_path (str): Local requirements file path (text file).
+    Returns:
+        Tuple[List[str], List[str]] - Packages and imports.
+    """
+    packages: List[str] = []
+    imports: List[str] = []
+    with open(file_path) as f:
+        for line in f:
+            line = line.strip()
+            if line and len(line) > 0:
+                if os.path.exists(line):
+                    imports.append(line)
+                else:
+                    packages.append(line)
+    return packages, imports
+
+
+def parse_conda_environment_yaml_file(
+    file_path: str,
+) -> Tuple[List[str], Optional[str]]:
+    """
+    Parses a Conda environment file (see https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#create-env-file-manually)
+    Python version passed in is used as the runtime version for sprocs/udfs.
+    Conda-style dependencies (numpy=1.2.3) are converted to pip-style dependencies (numpy==1.2.3).
+    Args:
+        file_path (str): Local requirements file path (yaml file).
+    Returns:
+        Tuple[List[str], Optional[str]] - Packages and Python runtime version, if specified. (Note that you cannot
+        specify local file or folder imports in a conda environment yaml file).
+    """
+    packages: List[str] = []
+    runtime_version: Optional[str] = None
+    with open(file_path) as f:
+        try:
+            environment_data = yaml.safe_load(f)
+            dependencies = environment_data.get("dependencies", [])
+            for dep in dependencies:
+                if isinstance(dep, str):
+                    dep = dep.strip()
+                    if any(r in dep for r in (">", "<")):
+                        raise ValueError(
+                            f"Conda dependency with ranges '{dep}' is not allowed! Please specify a single version."
```

Comment:
nit: Maybe we could say "not supported"? "not allowed!" feels very strong and seems like user's error 😂